src/drivers/ports/stm32f1/uart_port.h

```diff
@@ -31,7 +31,44 @@
 #ifndef __DRIVERS_UART_PORT_H__
 #define __DRIVERS_UART_PORT_H__
 
-#define UART_PORT_FRAME_FORMAT_DEFAULT 0
+/*
+ * STM32F1xx product line supports 8/9 data bits, N/E/O parity, 1/2 stop bits
+ * NOTE format field codification:
+ *      bits 0-7 are 8-15 of CR1 (coded 8 bits to the right)
+ *      bits 8-15 are 8-15 of CR2
+ */
+#define STM32_UART_DATA_BITS_8  0x00
+#define STM32_UART_DATA_BITS_9  ((0x01 << STM32_USART_CR1_M) >> 8)
```

Comment:
``STM32_USART_CR1_M`` is defined as ``BIT(12)``, which expands to ``1 << 12``, or ``4096`` decimal. So the compilation error persists selecting a non-default frame format. Please fix the compilation error for all frame formats.